### ISSUE TITLE
Switch from tagging releases as met-X.Y to using vX.Y instead.

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/MET#1541]
## Describe the Task ##
The existing MET build/release conventions are not consistent with the other METplus components. This task is to make the following changes to make it consistent:

1. Update the DockerHub automated build rules from expecting tags with a "met-" prefix to simply a "v" prefix.
2. Retag "met-9.1" as "v9.1" and "met-10.0_beta1" as "v10.0-beta1" and make sure DockerHub builds them correctly.
3. Update and test the build script to ensure that a tag named "v9.1" results in a tarfile named "met-9.1.YYYYMMDD.tar.gz" and a top-level directory named "met-9.1". So the script needs to strip off the leading "v".
4. Coordinate with METplus engineers to make sure that this doesn't break the TravisCI automated testing.

### Time Estimate ###
4 hours.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
None required.

### Relevant Deadlines ###
Prior to the met-10.0-beta2 release.

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required: John HG
- [x] Select **scientist(s)** or **no scientist** required: no scientist required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by JohnHalleyGotway** [comment]
I updated the DockerHub autobuild rules with:
1. DockerHub latest is built from GitHub tag vX.Y
2. DockerHub X.Y* is built from GitHub tag vX.Y* (includes vX.Y-betaN) 
3. DockerHub X.Y-latest is built from GitHub tag vX.Y or vX.Y.Z
I left the corresponding "met-" rules in place. Need to remember to delete them once the switch is done.

I recreated all the old tags using:
```
 git tag v9.1 met-9.1
 git push -u origin v9.1
```
Then I updated all the releases on GitHub to switch the tag from "met-9.1" to "v9.1". And lastly, I deleted all the old tags:
```
 git tag -d met-9.1
 git tag -u origin :met-9.1
```

I tested recreating the tarfile by running:
```
scripts/met_checkout_and_build.sh tag v9.1
```
But that creates "met-v9.1.20201027.tar.gz" which contains a "met-v9.1" directory. I need to update that script to strip the "v" from the "v9.1" in both cases.
